New mids2spss

gerkovink opened the pull request:
This PR would add the following:

- [x] New `mids2spss()` function that replaces the old `mids2spss()` generates one of the following:

  - a `.sav` data set that can be treated as a multiple imputed data set in SPSS
  - a `.zsav` compressed data set that can be treated as a multiple imputed data set in SPSS

- [x] Faster processing, both in `mice` and in SPSS
- [x] The new function deprecates `miceWriteForeignSPSS()`
- [x] Dependency `haven` is added to facilitate im/exporting data sets. This would make it more straightforward to facilitate the export of `mice` objects` to SPSS, SAS and Stata. It does add a dependency, though.

stefvanbuuren commented:
Gerko, thanks a lot. `miceWriteForeignSPSS()` was outdated, so it's great to see this update.

One request: Could you move `haven` to `suggests`? This evades adding a dependency, but assumes that `haven()` is part of the user library. In other to guarantee that, call `install.on.demand("haven")` at the top  of the function.

stefvanbuuren reviewed (changes requested):
See message

gerkovink commented:
- [x] Moved `haven` to suggests
- [x] Added `install.on.demand("haven")

New R CMD build check results:

```r
── R CMD check results ───────────────────────────────── mice 3.12.0 ────
Duration: 2m 55s

0 errors ✓ | 0 warnings ✓ | 0 notes ✓

R CMD check succeeded
```

stefvanbuuren commented:
Thanks.
